charset_normalizer/models.py

```diff
@@ -225,12 +225,10 @@ def has_submatch(self) -> bool:
     def alphabets(self) -> List[str]:
         if self._unicode_ranges is not None:
             return self._unicode_ranges
-        detected_ranges = set()  # type: Set[str]
-        for character in str(self):
-            detected_range = unicode_range(character)  # type: Optional[str]
-            if detected_range:
-                detected_ranges.add(detected_range)
-        self._unicode_ranges = sorted(list(detected_ranges))
+        # list detected ranges
+        detected_ranges = [unicode_range(char) for char in str(self)]
+        # filter and sort
+        self._unicode_ranges = sorted(filter(None.__ne__, detected_ranges))
```

Comment:
If the idea is to just filter out falsy values, `filter(None, ...)` is the idiom.

> If function is `None`, the identity function is assumed, that is, all elements of iterable that are false are removed.

Reply:
I just changed it because of mypy.